### DISCUSSION TITLE
Unable to set arbitrary value for a background

### DISCUSSION THREAD

**Discussion by dokgu** [tailwindlabs/tailwindcss#11766] [Help] [answered]
I have a React web app where you can create rooms and assign custom colors to them (hex), this color is then shown as a small circle beside the name of the room.

According to [this documentation](https://tailwindcss.com/docs/background-color#arbitrary-values):

> Arbitrary values
> 
> If you need to use a one-off color value that doesn’t make sense to include in your theme, use square brackets to generate a property on the fly using any arbitrary value.

```html
<p class="bg-[#50d71e]">
  <!-- ... -->
</p>
```

I did the same thing when displaying the color beside the room's name:

```html
<span
  className={`bg-[#${room.bgColor}] mr-4 h-2.5 w-2.5 rounded-full`}
  aria-hidden='true'
/>
<span className='truncate'>{room.name}</span>
```

When I inspect the page, this is the resulting HTML:

```html
<span class="bg-[#f43f5e] mr-4 h-2.5 w-2.5 rounded-full" aria-hidden="true"></span>
<span class="truncate">Room Name</span>
```

But somehow the color is not showing up. I only see the name of the room but no colored circle beside it. Using something like `bg-green-500` seems to show the circle but I don't want to use predefined colors from the TailwindCSS theme.

From further research - it looks like my issue is because of [dynamic class names](https://tailwindcss.com/docs/content-configuration#dynamic-class-names). However I can't know beforehand what the color to be selected is. Basically I have a color picker show up when creating a room and the user selects a color from it, the hex then gets saved on the database alongside the room's name. When I display the list of rooms, the color is coming in from the database and I can't pre-build the colors.

Is what I'm trying to achieve not possible with TailwindCSS?

**Comment by wongjn** [comment, marked as answer]
What you are trying to achieve is not possible with TailwindCSS. Consider using an inline `style` instead:

```jsx
<span className="mr-4 h-2.5 w-2.5 rounded-full" aria-hidden="true" style={{ backgroundColor: `#${room.bgColor}` }}></span>
<span className="truncate">Room Name</span>
```

**Reply by dokgu** [reply]
Thank you - seems like a very simple solution. Hopefully someday TailwindCSS can also support values that aren't pre-built.
